activestorage/lib/active_storage/service/disk_service.rb

```diff
@@ -9,10 +9,10 @@ module ActiveStorage
   # Wraps a local disk path as an Active Storage service. See ActiveStorage::Service for the generic API
   # documentation that applies to all services.
   class Service::DiskService < Service
-    attr_reader :root
+    attr_reader :root, :host
 
-    def initialize(root:)
-      @root = root
+    def initialize(root:, host:)
+      @root, @host = root, host
```

Comment:
Why does a disk service need a host?  It seems like a disk service wouldn't be over a network, so I don't understand this change.  Am I missing something?

Reply:
It needs a host for the purpose of generating URLs.

Reply:
After reading the diff more closely, it looks like this is so that people can change the host for the generated URLs.  Maybe we should default the host when loading the config?  The reason I'm asking is because this change broke some existing configurations (ones without `host` specified).

Also we should probably fix the exception such that the filename for the malformed configuration is in the exception.

Reply:
> it looks like this is so that people can change the host for the generated URLs

It’s because generated URLs previously didn't have any host at all. (`ActiveStorage::Service::DiskService#service_url` always returned a path.)  `ActiveStorage::Identification` (reasonably) expects `#service_url` to return a valid URL.

Reply:
I apologize for not including this in the changelog, by the way. In retrospect, it deserved mention there.

Reply:
> I’m not sure what default we’d use. How would we know what host to use at boot time?

It looks like all of the generators generate `http://localhost:3000` as the default (which, technically isn't a host, but a URI).  Maybe that's a good start?  This code will give you the hostname of the current system:

```
$ ruby -rsocket -e'puts Socket.gethostname'
```

Since this is for storing / serving files locally, I'd expect localhost/3000 to be fine defaults.  Is there a reason why they aren't?

> I apologize for not including this in the changelog, by the way. In retrospect, it deserved mention there.

No problem!! 🤗

Reply:
> It looks like all of the generators generate `http://localhost:3000` as the default (which, technically isn't a host, but a URI). Maybe that's a good start?

Sure. I’ll make that change in a bit.

Reply:
Added a default in 6fb3ac1.